### PR TITLE
Detach patterns and licenses

### PR DESCRIPTION
**Pull request by coolo**
This is a rather large PR - and actually still a lot left to do (e.g. in editing patterns and I'm not even sure the tests run at this very moment).

The main changes:
 - patterns have no more licence id, they are only grouped by their license name. Next step is removing licenses from the UI all together. But we need a proper replacment before we can do that.
 - ignore patterns without licences in the report, but expand snippets instead. Either unclassified or classified as license.

Example from staging area:

Before classification: 
screencapture-localhost-3000-reviews-details-1-2019-07-09-17_34_51

After classification:
screencapture-localhost-3000-reviews-details-1-2019-07-09-17_38_06

Note the classifier removed quite some real matches - because it wasn't trained on artistic too much, because most of it is covered my license patterns in production.

### CODE REVIEW

File: migrations/cavil.sql [hunk 1]
```diff
@@ -215,3 +215,31 @@ create index on snippets(approved);
...
+
+-- 9 down
+alter table license_patterns alter column license_string drop default;
+alter table license_patterns alter column license_string set null;
+
+-- 10 up
+alter table license_patterns rename column license_string to license;
+
+-- 10 down
+alter table license_patterns rename column license to license_string;
```
Comment (kraih) [inline migrations/cavil.sql]: You probably should make those one migration before merging. 😉 It doesn't matter if 7, 8 and 9 are not present as long as the latest number is still there, so those could just be merged into 6 for example.

File: lib/Cavil/Bootstrap.pm [hunk 2]
```diff
@@ -28,27 +28,30 @@ sub load {
   # Avoid adding the same data more than once
   my $licenses = $self->app->licenses;
   return undef
-    if $licenses->pg->db->select('licenses', 'id', {name => 'Low Risk Keyword'})
+    if $licenses->pg->db->select('licenses', 'id', {name => 'Apache-2.0'})
```
Comment (kraih) [inline lib/Cavil/Bootstrap.pm]: Was this change intentional?

File: migrations/cavil.sql (hunk 1 above)
Comment (coolo) [reply migrations/cavil.sql]: I need to migrate the data before dropping the license key. So I can't have them in one migration step.

File: lib/Cavil/Bootstrap.pm (hunk 2 above)
Comment (coolo) [reply lib/Cavil/Bootstrap.pm]: Yes. there is no license 'low risk keyword' anymore.
